In [1]:
import pandas as pd
import requests
from bs4 import BeautifulSoup  # beautifulSopu library is ussed for webscapping 

In [2]:
requests.get('https://www.ambitionbox.com/list-of-companies?page=1')

<Response [403]>

# response 403 means bad request means server rejected the request

In [3]:
requests.get('https://www.ambitionbox.com/list-of-companies?page=1').text

'<HTML><HEAD>\n<TITLE>Access Denied</TITLE>\n</HEAD><BODY>\n<H1>Access Denied</H1>\n \nYou don\'t have permission to access "http&#58;&#47;&#47;www&#46;ambitionbox&#46;com&#47;list&#45;of&#45;companies&#63;" on this server.<P>\nReference&#32;&#35;18&#46;c68c2c31&#46;1765472978&#46;1296eb43\n<P>https&#58;&#47;&#47;errors&#46;edgesuite&#46;net&#47;18&#46;c68c2c31&#46;1765472978&#46;1296eb43</P>\n</BODY>\n</HTML>\n'

# here in the text access denied is shown means we don't have the permission to access this page

In [4]:
headers={'User-Agent':'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.36'}
webpage = requests.get('https://www.ambitionbox.com/list-of-companies?page=1',headers=headers).text

In [5]:
 soup = BeautifulSoup(webpage , 'lxml')  # lxml is html traversal used for traversing html data it is mandatory to give it to BeaitifulSoup .

In [34]:
# soup.prettify()    

# soup.prettify() method formats the html documnet and content we can see that by printing it like this print(soup.prettify())

In [7]:
soup.find_all('h1')[0].text

'\n\t\t\t\t\tTop Companies in\n\t\t\t\t \n\t\t\t\t\tINDIA\n\t\t\t\t\t'

In [9]:
for i in soup.find_all('h2'):
    print(i.text.strip())     # here strip() is used for filling the gaps and formats the data

Companies in India
TCS
Accenture
Wipro
Cognizant
Capgemini
HDFC Bank
Infosys
ICICI Bank
HCLTech
Tech Mahindra
Genpact
Teleperformance
Axis Bank
Jio
Concentrix Corporation
Amazon
iEnergizer
Reliance Retail
IBM
LTIMindtree
Popular Collections by Industries
Popular Collections by Cities
Popular Collections by Roles


In [15]:
company = soup.find_all('div',class_ = 'companyCardWrapper')

In [16]:
len(company)

20

# main block's will be taken here and printed here all the company names

In [25]:
name = []
for i in company:
    name.append(i.find('h2').text.strip())

In [26]:
name

['TCS',
 'Accenture',
 'Wipro',
 'Cognizant',
 'Capgemini',
 'HDFC Bank',
 'Infosys',
 'ICICI Bank',
 'HCLTech',
 'Tech Mahindra',
 'Genpact',
 'Teleperformance',
 'Axis Bank',
 'Jio',
 'Concentrix Corporation',
 'Amazon',
 'iEnergizer',
 'Reliance Retail',
 'IBM',
 'LTIMindtree']

In [43]:
rating = soup.find_all('div' , class_ = 'rating_text rating_text--md ')
rating2 = []
for  i in company:
    rating2.append(i.find_all('text', class_ = 'rating_text rating_text--md '))

In [44]:
rating2

[[],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 [],
 []]